pkg/code-generator/codegen/conversion.go

```diff
@@ -0,0 +1,165 @@
+package codegen
+
+import (
+	"bytes"
+	"sort"
+	"text/template"
+
+	"github.com/solo-io/go-utils/versionutils/kubeapi"
+	"github.com/solo-io/solo-kit/pkg/errors"
+
+	"github.com/solo-io/go-utils/log"
+	code_generator "github.com/solo-io/solo-kit/pkg/code-generator"
+	"github.com/solo-io/solo-kit/pkg/code-generator/codegen/templates"
+	"github.com/solo-io/solo-kit/pkg/code-generator/model"
+)
+
+func GenerateConversionFiles(soloKitProject *model.ApiGroup, projects []*model.Version) (code_generator.Files, error) {
+	var files code_generator.Files
+
+	sort.SliceStable(projects, func(i, j int) bool {
+		vi, err := kubeapi.ParseVersion(projects[i].VersionConfig.Version)
+		if err != nil {
+			return false
+		}
+		vj, err := kubeapi.ParseVersion(projects[j].VersionConfig.Version)
+		if err != nil {
+			return false
+		}
+		return vi.LessThan(vj)
+	})
+
+	resourceNameToProjects := make(map[string][]*model.Version)
+
+	for index, project := range projects {
+		for _, res := range project.Resources {
+			// only generate files for the resources in our group, otherwise we import
+			if !project.VersionConfig.IsOurProto(res.Filename) && !res.IsCustom {
```

Comment:
Nit: very similar logic is done in a few places, e.g. code-generator. Maybe can be combined into a function